Libraries/LibWeb/HTML/HTMLFormElement.cpp

```diff
@@ -561,11 +561,49 @@ unsigned HTMLFormElement::length() const
     return elements()->length();
 }
 
+// https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#statically-validate-the-constraints
+HTMLFormElement::StaticValidationResult HTMLFormElement::statically_validate_contraints()
+{
+    // 1. Let controls be a list of all the submittable elements whose form owner is form, in tree order.
+    auto controls = get_submittable_elements();
+    // 2. Let invalid controls be an initially empty list of elements.
+    Vector<GC::Ref<DOM::Element>> invalid_controls;
+    // 3. For each element field in controls, in tree order:
+    for (auto& element : controls) {
+        VERIFY(is<FormAssociatedElement>(*element));
+        auto& field = dynamic_cast<FormAssociatedElement&>(*element);
```

Comment:
These 2 lines can be simplified to `auto& field = as<FormAssociatedElement>(*element);`